aws_lambda_powertools/event_handler/api_gateway.py

```diff
@@ -709,7 +709,7 @@ class ResponseBuilder(Generic[ResponseEventT]):
     def __init__(
         self,
         response: Response,
-        serializer: Callable[[Any], str] = json.dumps,
+        serializer: Callable[[Any], str] = partial(json.dumps, separators=(",", ":"), cls=Encoder),
```

Comment:
this simplifies implementation, as we move serialization logic to our Json Encoder instead.